### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by didierganthier**
## Checklist
- [x] I have read this repository's contribution guidelines.
- [x] My pull request has a descriptive title (not a vague title like Update index.MD)
- [x] My pull request targets the main branch of this [project](https://github.com/didierganthier/my_blog).
- [x] I added relevant documentation for all the files I changed

## Summary
This PR adds authentication 

 ## What changes have been made?
- [x] Register a new user.
- [x] User logs in with a combination of email and password.
- [x] Hashed password is stored in the database.
- [x] Ask for confirmation of email.
- [x] Can reset password.

## Credentials
- email: didierganthierperan@gmail.com
- password: anything

## Authors
- @didierganthier
- @rbreva

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Sign in", class: "bg-blue-400 text-white py-2 px-4 hover:bg-blue-600 rounded cursor-pointer" %>
+  </div>
+  <div class="py-4">
+    <%= render "devise/shared/links" %>
+  </div>
+<% end %>
```
Comment (CollinsTatang) [inline app/views/devise/sessions/new.html.erb]: - [ ] Please modified your login view style to match the given [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md)

blog_login

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (CollinsTatang) [inline app/models/user.rb:3]: Please enable `:confirmable ` for your application to be able to asks for confirmation of email.

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @didierganthier , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brytebee — APPROVED** [review]
Hi, @didierganthier & @CollinsTatang👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



##

###  **Optional Suggestions:**
- N/A

##

Good luck! 🎉 

------

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**